clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -165,11 +165,69 @@ def CIR_ArrayType : CIR_Type<"Array", "array",
   let hasCustomAssemblyFormat = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// FuncType
+//===----------------------------------------------------------------------===//
+
+def CIR_FuncType : CIR_Type<"Func", "func"> {
+  let summary = "CIR function type";
+  let description = [{
+    The `!cir.func` is a function type. It consists of a single return type, a
+    list of parameter types and can optionally be variadic.
+
+    Example:
+
+    ```mlir
+    !cir.func<!bool ()>
+    !cir.func<!s32i (!s8i, !s8i)>
+    !cir.func<!s32i (!s32i, ...)>
+    ```
+  }];
+
+  let parameters = (ins ArrayRefParameter<"Type">:$inputs, ArrayRefParameter<"Type">:$results,
+                        "bool":$varArg);
```

Comment:
Perhaps align the second argument in the second line here?

Reply:
Applied.